### ISSUE TITLE
Billing question

### ISSUE THREAD

**Issue by russdirks** [vipnode/vipnode.org#28]
As a light client, am I billed per connected host, or just flat rate per minute connected to the pool?  I had originally thought a light client only latched onto one host, but I've been  monitoring net.peerCount on my light client and am seeing it sometimes go up to 4-7 peers.  Do I get billed for each one of those?

**Comment by shazow** [comment]
You're billed for every host in the pool that you're peered with. If you run with -vv logging, it will show up as the number of "active" peers.

Keep in mind, the official vipnode.org pool is just a demo pool, it doesn't require any real money and it will happily serve you even if your balance is 0. :)

You can tell the agent how many pool peers you want using  `vipnode agent --min-peers=3`.

Let me know if you have any other questions!

**Comment by mining-visualizer** [comment]
Oh, ok.  Thanks for the info.  I've been running vipnode for about 2 days now, and I don't think I've ever seen "active" be anything other than zero, and yet my light node, which I ran with `--nodiscover`, seems to be connected and downloading blocks.  Comments?
Also, I noticed "peers" seems to correlate with what I see running net.peerCount.  Where are those peers coming from? Why are they not "active"?  I did not specify `--min-peers` which I understand defaults to 3.

**Comment by shazow** [comment]
@mining-visualizer Are you on v2.2.1? We might need to wait until more hosts are on v2.2.1 also (previous versions weren't accepting the matchmaking properly).

Even with nodiscover, geth will still discover more peers through connected peers. I think nodiscover just disables the "bootstrapping" process (finding the initial set of peers to connect to). So it's totally possible that you discover your first peer via vipnode, then your node discovers more peers from that peer, then you disconnect from the original peer.

**Comment by mining-visualizer** [comment]
Oh, I see, thanks.  That answers a few other questions I had.  Yes, I'm on v2.2.1.  So you're not running any hosts yourself with up-to-date software?  Have you considered putting an expiry feature in the software to force people to update every so often, especially since it is still in beta?
I think this is a very cool project.  You might not be seeing a lot of activity now, but it definitely could become more of a thing now that Infura has started charging for the higher levels of service.

**Comment by shazow** [comment]
> So you're not running any hosts yourself with up-to-date software?

I used to, but we have enough community nodes that I cut back for now. Most of the nodes are being run thanks to DAppNode--I reached out to them a while ago, they should be updating their nodes soon. I should toss in another Geth node in there or two, though! :thinking: 

> Have you considered putting an expiry feature in the software to force people to update every so often, especially since it is still in beta?

I don't consider it beta since v2.0. Old versions weren't advertising the agent version to the pool, but recent versions added that, so in a future version I can send a warning to outdated agents that they need to update. Hopefully that will help!

> I think this is a very cool project. You might not be seeing a lot of activity now, but it definitely could become more of a thing now that Infura has started charging for the higher levels of service.

Yea, I think a big part of it is also whenever Ethereum gets hyped and there's an influx of new users. Whenever there's another big wave of adoption, I think vipnode will be very useful. Meanwhile, Infura is actually funding my work on this project, so it's working out for everyone. :)
